server/src/storage/datastore.ts

```diff
@@ -33,4 +33,15 @@ const getWithId = async (kind: string, id: string) => {
   };
 };
 
-export {getWithId};
+/**
+ * A Datastore wrapper that inserts a particular entity with the specified Kind.
+ * @param kind The Kind of the Entity
+ * @param data The data of the Entity to be added
+ */
+const add = async (kind: string, data: object) => {
```

Comment:
How will you prevent same data to be added multiple times?
Let's say merchant clicks 'sign up' button twice with the same form data

Reply:
I haven't added checks for that yet. Was planning to do it in another PR. 

Anyway, my plan is to make sure that the firebase `uid` is unique in the table. Users with different `uid` will have different `email` as ensured by Firebase Authentication. I know that for SQL databases we can put this restriction on the database. I'm hoping there is a similar way to do this for Datastore but I need to find out more on that. But even with that, if we want to be defensive we can still add a code to check.